### DISCUSSION TITLE
mouse and keyboard control not possible when connecting intel to intel mac one mac running macOS 12 Monterey

### DISCUSSION THREAD

**Discussion by foss-** [rustdesk/rustdesk#4871] [General]
### Bug Description

There are some outstanding issues which are either happening on intel mac to intel mac connections and or when one of the macs is running macOS 12 Monterey. Don't have enough macs to test and figure out with certainty which of the two is causing the trouble.

Was able to successfully connect intel mac running Ventura to Apple Silicon mac in both directions and use keyboard and mouse as expected.

Issues encountered with rustdesk 1.2.0

Intel mac Ventura 13.4.1 to Intel mac Monterey 12.6.7:

* ✅ image shows, ❌ no mouse control, ❌ no keyboard control

Intel mac Monterey 12.6.7 to Intel mac Ventura 13.4.1:

* ✅ image shows, ✅ mouse control, ❌ no keyboard control

When testing Intel mac Ventura 13.4.1 to M1 mac Ventura 13.4.1 everything was working as expected in both directions. So this feels as if it is probably caused by macOS 12 Monterey.

### How to Reproduce

See above

### Expected Behavior

Mouse and keyboard control even when interacting with macs running macOS 12 Monterey.

### Operating system(s) on local side and remote side

macOS 13.4.1 <-> macOS 12.6.7

### RustDesk Version(s) on local side and remote side

1.2.0

### Screenshots

N/A

**Comment by foss-** [comment]
Since [another commit](https://github.com/rustdesk/rustdesk/commit/a96eb236c1c7b3d62fe8a819ba861f54921021e9) trying to resolve a permission situation on macOS is part of 1.2.1, retested on macOS 13.4.1.
- removed RustDesk from `System Settings > Privacy & Security > Input Monitoring`
- re-opened RustDesk
- was greeted with Permissions info recommending to install background service
- added that and was informed `Background Items Added
Software from "zhou huabing" added items that can run in the background. You can manage this in Login Items Settings.`

Curiously RustDesk is not requesting additional permissions despite me removing the Input Monitoring permission. I would think that is unexpected behavior. When I double checked `Input Monitoring` permissions, there was no RustDesk entry (as expected). However I am wondering, shouldn't RustDesk be requesting this permission then?

Briefly tested this on macOS Monterey 12.6.7 mac of a friend and there the permission dialog never vanished even after removing and re-adding both Login Item and Input Monitoring Permission. After re-adding both and restarting RustDesk 1.2.1 it still is requesting Permissions. When trying to grant the requested permission, System Settings > Privacy and Security > Accessibility Controls` but RustDesk is already on the list and enabled.

No matter the permissions I am still unable to control the remote macOS 12 mac with my mouse. https://github.com/rustdesk/rustdesk/discussions/4871

Unrelated: Would renaming the login item to `RustDesk` be possible?

**Comment by zhweiGuo** [comment]
Try removing and re-adding these 3 permissions of Rustdesk1.2.1.
I re-controled mac12.4 by Ubuntu22.04.
![image](https://github.com/rustdesk/rustdesk/assets/15942478/18a3bba1-500f-4bca-8215-ddcd53c73083)

**Comment by foss-** [comment]
We've removed Rustdesk permissions on macOS 12.6.8 under System Settings > Security >

- Accessibility
- Input Monitoring
- Screenrecording

Restarted macOS, re-add the three Rustdesk permissions. Restart Rustdesk, no hint about permissions shows up so it seems that worked fine. And indeed, mouse and keyboard control work again as expected 🙌
